src/database-sqlite3.h

```diff
@@ -30,37 +32,99 @@ extern "C" {
 class Database_SQLite3 : public Database
 {
 public:
-	Database_SQLite3(const std::string &savedir);
-	~Database_SQLite3();
+	virtual ~Database_SQLite3();
 
 	void beginSave();
 	void endSave();
 
-	bool saveBlock(const v3s16 &pos, const std::string &data);
-	void loadBlock(const v3s16 &pos, std::string *block);
-	bool deleteBlock(const v3s16 &pos);
-	void listAllLoadableBlocks(std::vector<v3s16> &dst);
 	bool initialized() const { return m_initialized; }
+protected:
+	Database_SQLite3(const std::string &savedir, const std::string &dbname);
 
-private:
-	// Open the database
-	void openDatabase();
-	// Create the database structure
-	void createDatabase();
 	// Open and initialize the database if needed
 	void verifyDatabase();
 
-	void bindPos(sqlite3_stmt *stmt, const v3s16 &pos, int index = 1);
+	// Convertors
+	inline void str_to_sqlite(sqlite3_stmt *s, const int iCol, const std::string &str) const
+	{
+		sqlite3_vrfy(sqlite3_bind_text(s, iCol, str.c_str(), str.size(), NULL));
+	}
+
+	inline void str_to_sqlite(sqlite3_stmt *s, const int iCol, const char *str) const
+	{
+		sqlite3_vrfy(sqlite3_bind_text(s, iCol, str, strlen(str), NULL));
+	}
+
+	inline void int_to_sqlite(sqlite3_stmt *s, const int iCol, const int val) const
+	{
+		sqlite3_vrfy(sqlite3_bind_int(s, iCol, val));
+	}
+
+	inline void int64_to_sqlite(sqlite3_stmt *s, const int iCol, const s64 val) const
+	{
+		sqlite3_vrfy(sqlite3_bind_int64(s, iCol, (sqlite3_uint64)val));
+	}
+
+	inline void double_to_sqlite(sqlite3_stmt *s, const int iCol, const double val) const
+	{
+		sqlite3_vrfy(sqlite3_bind_double(s, iCol, val));
+	}
+
+	inline std::string sqlite_to_string(sqlite3_stmt *s, int iCol)
+	{
+		const char* text = reinterpret_cast<const char*>(sqlite3_column_text(s, iCol));
+		return std::string(text ? text : "");
+	}
+
+	inline s32 sqlite_to_int(sqlite3_stmt *s, int iCol)
+	{
+		return sqlite3_column_int(s, iCol);
+	}
+
+	inline u32 sqlite_to_uint(sqlite3_stmt *s, int iCol)
```

Comment:
sqlite only have int types, int_to_sqlite is used when writing inventory and an implicit cast is done. For the loading this function permit to offer and implicit cast to retrieve unsigned int for inventories